Src/FluentAssertions/CallerIdentification/QuotesHandler.cs

```diff
@@ -0,0 +1,53 @@
+﻿using System.Text;
+
+namespace FluentAssertions.CallerIdentification
+{
+    public class QuotesHandler : IHandler
+    {
+        private readonly StringBuilder statement;
+        private char isQuoteEscapeSymbol = '\\';
+        private bool isQuoteContext;
+        private char? previousChar;
+
+        public QuotesHandler(StringBuilder statement) => this.statement = statement;
+
+        public HandlerResult Handle(char symbol)
+        {
+            if (symbol == '"')
+            {
+                if (isQuoteContext)
+                {
+                    if (previousChar != isQuoteEscapeSymbol)
+                    {
+                        isQuoteContext = false;
+                        isQuoteEscapeSymbol = '\\';
+                    }
+                }
+                else
+                {
+                    isQuoteContext = true;
+                    if (IsVerbatim())
+                    {
+                        isQuoteEscapeSymbol = '"';
+                    }
+                }
+            }
+
+            if (isQuoteContext)
+            {
+                statement.Append(symbol);
+            }
+
+            previousChar = symbol;
+            return isQuoteContext ? HandlerResult.Handled : HandlerResult.InProgress;
+        }
+
+        private bool IsVerbatim()
+        {
+            return previousChar == '@'
+               || (statement.Length > 1
+                   && statement[statement.Length - 1] == '$'
```

Comment:
🔧 Isn't it clearer to use `EndsWith("$@")` or use the new C# range syntax?

Reply:
Both of those suggestions sound great, but 1 requires changing the framework, and the other (`EndsWith` doesn't work on `StringBuilder`, and calling `ToString` kind of defeats the purpose).

Maybe we leave as is?